tests/cephfs/clients/verify_root_squash_in_caps.py

```diff
@@ -0,0 +1,215 @@
+                new_client_hostname="add_data",
+                extra_params=f" --client_fs {fs_name}",
+            )
+            cmd = f"mount > {mnt_path}/{filename};mount > {mnt_path}/volumes/{filename}"
+            client.exec_command(
+                sudo=True,
+                cmd=cmd,
+                check_ec=False,
+            )
+            cmd = f"mkdir {mnt_path}/{dirname};mkdir {mnt_path}/volumes/{dirname}"
+            client.exec_command(sudo=True, cmd=cmd)
+            client.exec_command(sudo=True, cmd=f"umount {mnt_path}")
+            client.exec_command(sudo=True, cmd="ceph auth del client.add_data")
+
+        client_name = "client.verify_root_squash"
+        mds_cap_fs1 = f"'allow rw fsname={fs_list[0]} root_squash, allow rw fsname={fs_list[0]} path=/volumes"
+        mds_cap_fs2 = f",allow rw fsname={fs_list[1]} root_squash, allow rw fsname={fs_list[1]} path=/volumes'"
+        mon_cap_fs = f"'allow r fsname={fs_list[0]},allow r fsname={fs_list[1]}'"
+        osd_cap_fs = f"'allow rw tag {fs_list[0]} data={fs_list[0]},allow rw tag {fs_list[1]} data={fs_list[1]}'"
+        caps = f"mds {mds_cap_fs1}{mds_cap_fs2} mon {mon_cap_fs} osd {osd_cap_fs}"
+        create_cmd = f"ceph auth get-or-create {client_name} {caps}"
+        log.info(f"Set client user {client_name} with caps for root squash")
+        client.exec_command(sudo=True, cmd=create_cmd)
+        key_ring_path = f"/etc/ceph/ceph.{client_name}.keyring"
+        get_cmd = f"ceph auth get {client_name} -o {key_ring_path}"
+        client.exec_command(sudo=True, cmd=get_cmd)
+        out, rc = client.exec_command(sudo=True, cmd=f"cat {key_ring_path}")
+        if "root_squash" not in out:
+            raise CommandFailed(f"Failed to add root_squash cap to {client_name}")
+        log.info(
+            f"ceph keyring file for {client_name} with root_squash enabled : \n\n{out}"
+        )
+
+        log.info("Verify root_squash cap works on kernel and fuse mount clients")
+        mount_path = {}
+        for fs_name in fs_list:
+            mount_path[fs_name] = {}
+            mount_path[fs_name]["kernel"] = f"/mnt/{fs_name}_kernel_{mounting_dir}_1/"
+            mount_path[fs_name]["fuse"] = f"/mnt/{fs_name}_fuse_{mounting_dir}_1/"
+        mon_node_ips = fs_util.get_mon_node_ips()
+        for fs_name in fs_list:
+            client.exec_command(
+                sudo=True, cmd=f"mkdir -p {mount_path[fs_name]['kernel']}"
+            )
+            fs_util.kernel_mount(
+                [client],
+                mount_path[fs_name]["kernel"],
+                ",".join(mon_node_ips),
+                new_client_hostname="verify_root_squash",
+            )
+            mon_ips = ",".join(mon_node_ips)
+            mount_cmd = f"mount -t ceph {mon_ips}:/ {mount_path[fs_name]['kernel']} "
+            mount_cmd += f"-o name=verify_root_squash,secretfile=/etc/ceph/verify_root_squash.secret,fs={fs_name}"
+            client.exec_command(sudo=True, cmd=mount_cmd)
```

Comment:
From line no: 109 t0 112 be replaced with below kernel mount call
```
fs_util.kernel_mount(
                [client],
                mount_path[fs_name]["kernel"],
                ",".join(mon_node_ips),
                new_client_hostname="verify_root_squash",
                extra_params=f",fs={fs_name}",
            )
```

Reply:
added